pydantic_ai_slim/pydantic_ai/output.py

```diff
@@ -0,0 +1,289 @@
+from __future__ import annotations
+
+from collections.abc import Awaitable, Sequence
+from dataclasses import dataclass
+from typing import Any, Callable, Generic, Literal, Union
+
+from typing_extensions import TypeAliasType, TypeVar, get_args, get_origin
+from typing_inspection import typing_objects
+from typing_inspection.introspection import is_union_origin
+
+from .messages import RetryPromptPart
+from .tools import RunContext
+
+OutputDataT = TypeVar('OutputDataT', default=str, covariant=True)
+"""Covariant type variable for the result data type of a run."""
+
+T = TypeVar('T')
+
+OutputMode = Literal['text', 'tool', 'model_structured', 'prompted_structured', 'tool_or_text']
+"""All output modes."""
+StructuredOutputMode = Literal['tool', 'model_structured', 'prompted_structured']
+"""Output modes that can be used for structured output. Used by ModelProfile.default_structured_output_mode"""
```

Comment:
do all these need to be public?

Let's add `__all__` and put anything that should be public in there, and try try to make everything else private with a leading underscore.

Reply:
They're all used by other files and as types of public fields on public classes defined here and elsewhere, so if we prefixed them with an underscore here we'd need `# pyright: ignore[reportPrivateUsage]` there, and users would still be exposed to them.